doc/source/dev/howto-docs.rst

```diff
@@ -215,6 +215,218 @@ Note that for documentation within NumPy, it is not necessary to do
 Please use the ``numpydoc`` :ref:`formatting standard <numpydoc:format>` as
 shown in their :ref:`example <numpydoc:example>`.
 
+.. _doc_c_code:
+
+Documenting C/C++ Code
+======================
+
+NumPy uses Doxygen_ to parse specially-formatted C/C++ comment blocks. This generates
+XML files, which are  converted by Breathe_ into RST, which is used by Sphinx.
+
+**It takes three steps to complete the documentation process**:
+
+1. Writing the comment blocks
+-----------------------------
+
+Although there is still no commenting style set to follow, the Javadoc
+is more preferable than the others due to the similarities with the current
+existing non-indexed comment blocks.
+
+.. note::
+   Please see `"Documenting the code" <https://www.doxygen.nl/manual/docblocks.html>`__.
+
+This is how Javadoc style looks like::
+
+  /**
+   * Your brief is here.
+   */
+  void function(void);
+
+For line comment, insert a triple forward slash::
+
+  /// Your brief is here.
+  void function(void);
+
+And for structured data members, just insert triple forward slash and less-than sign::
+
+  /// Data type brief description
+  typedef struct {
+      int member_1; ///< Member 1 description.
+      int member_2; ///< Member 2 description.
+  } data_type;
+
+
+Common Doxygen Tags:
+++++++++++++++++++++
+
+.. note::
+   For more tags/commands, please take a look at https://www.doxygen.nl/manual/commands.html
+
+``@brief``
+
+Starts a paragraph that serves as a brief description. By default the first sentence
+of the documentation block is automatically treated as a brief description, since
+option `JAVADOC_AUTOBRIEF <https://www.doxygen.nl/manual/config.html#cfg_javadoc_autobrief>`__
```

Comment:
Is there a reason to prefer `JAVADOC_AUTOBRIEF` over requiring explicit `@brief` and `@details`?

Reply:
As far as I know, there's no way to force Doxygen to require `@brief` and `@details`. If `JAVADOC_AUTOBRIEF` is off it will never interpret parsing the brief. Why we should require `@brief and @details` in the first place?

Reply:
In other words, if JAVADOC_AUTOBRIEF is OFF, then it will follow the Qt style.